app/src/main/java/com/futurice/freesound/network/api/model/Sound.kt

```diff
@@ -28,51 +28,51 @@ import java.util.*
 @Fieldable
 data class Sound(
```

Comment:
The tricky thing we have here is that depending which field are actually requested from the backend, then the properties can actually be nullable from the API. Strictly speaking, there are no inherent non-nullable properties for `Sound`. In our current usage, we are using all the fields generated from `@Fieldable`, but there's no reason why another part of the API service layer could not make a request with a subset of those fields, in which case we would expect the field to be null.

Reply:
Perhaps we should allow the fields to be nullable in the API response mapped class (currently `Sound`) but then have another data type which is used by the remainder of the app.  We should rename `Sound` to be `SoundResponse` and then redefine a `Sound` class with non-nullable field for use by clients of the API service. Add a mapping layer which enforces the intended nullability.

Reply:
I think that would still fit within the scope of this PR.

Reply:
In the documentation, there is nothing mentioned about nullability.
That is why I've tried out the app with the current nullability setup and worked properly.
I have added a Timber logging to where we are passing them and we should have an eye on that via crashalitycs.
I would not like to add nullability just because we do not know

Reply:
When requesting a sound from the API you need to request the fields that you wish to be populated (Jonty and all that) - this means that in the JSON response any fields not requested are missing and hence would be null when mapped to `Sound.kt`.

Reply:
Ok, let's discuss this week the Fieldable and your idea

Reply:
@tomaszpolanski Let's progress with what we have and hopefully we have the problem of devising a new solution later.